File: webapp/frontend/views/uix/components/labeled_frame_with_button/labeled_frame_with_button.py
import os
import random
import tornado.web

from typing import List


class LabeledFrameWithButton(tornado.web.UIModule):
    
    def css_files(self):
        return [os.path.join("css", "labeled_frame_with_button.css")]

    def render(self, label:str, key:str, alt:str = "", title:str = ""):
        return self.render_string(
            "labeled_frame_with_button.html",
            label = label,
            icon_path= os.path.join("images", "delete-icon.webp"),
            key=key, 
            alt=alt,
            title = title)

File: webapp/frontend/views/uix/components/labeled_frame_with_button/labeled_frame_with_button.ts
import GenerateCircularIconButton from "../circular_icon_button/circular_icon_button.js"


class GenerateLabeledFrameWithButton
{
    #label:string 
    #key:string
    #alt:string=''
    #title:string=''
    constructor(label:string, key:string, alt:string, title:string)
    {
        this.#label = label
        this.#key = key
        this.#alt = alt
        this.#title = title
    }

    render()
    {
        return (`
        <div class="labeled-frame-with-button">
            <span>${this.#label}</span>
            ${(new GenerateCircularIconButton('./../static/images/delete-icon.webp', this.#key, this.#alt, this.#title)).render()}
        </div>
        `)
    }
}

export default GenerateLabeledFrameWithButton